### ISSUE TITLE
it doesn't work on many ontologies, both for obo and for owl formats. 

### ISSUE THREAD

**Issue by jac2130** [althonos/pronto#24]
#it seems as though it is quite sensitive to any errors in the owl code or obo representation? 

```python
import pronto
sd =pronto.Ontology('https://github.com/SDG-InterfaceOntology/sdgio/blob/master/sdgio.owl')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "build/bdist.linux-x86_64/egg/pronto/ontology.py", line 109, in __init__
  File "build/bdist.linux-x86_64/egg/pronto/ontology.py", line 224, in parse
  File "build/bdist.linux-x86_64/egg/pronto/utils.py", line 62, in new_func
  File "build/bdist.linux-x86_64/egg/pronto/parser/owl.py", line 63, in parse
  File "src/lxml/etree.pyx", line 3426, in lxml.etree.parse
  File "src/lxml/parser.pxi", line 1861, in lxml.etree._parseDocument
  File "src/lxml/parser.pxi", line 1881, in lxml.etree._parseFilelikeDocument
  File "src/lxml/parser.pxi", line 1776, in lxml.etree._parseDocFromFilelike
  File "src/lxml/parser.pxi", line 1187, in lxml.etree._BaseParser._parseDocFromFilelike
  File "src/lxml/parser.pxi", line 601, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 711, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 640, in lxml.etree._raiseParseError
  File "https://github.com/SDG-InterfaceOntology/sdgio/blob/master/sdgio.owl", line 44
lxml.etree.XMLSyntaxError: Specification mandate value for attribute data-pjax-transient, line 44, column 91
```

```python
envo = pronto.Ontology('https://raw.githubusercontent.com/EnvironmentOntology/envo/master/envo.obo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "build/bdist.linux-x86_64/egg/pronto/ontology.py", line 109, in __init__
  File "build/bdist.linux-x86_64/egg/pronto/ontology.py", line 224, in parse
  File "build/bdist.linux-x86_64/egg/pronto/parser/obo.py", line 70, in parse
  File "build/bdist.linux-x86_64/egg/pronto/parser/obo.py", line 236, in _classify
  File "build/bdist.linux-x86_64/egg/pronto/synonym.py", line 138, in from_obo
  File "build/bdist.linux-x86_64/egg/pronto/synonym.py", line 118, in __init__
ValueError: Undefined synonym type: synonym
```

and this one times out:
```python
gaz = pronto.Ontology('http://purl.obolibrary.org/obo/gaz.obo')
```

**Comment by althonos** [comment]
Unfortunately for the first one there is not much I can do since the error comes from the XML parser, meaning the `sdgio` is not a valid XML.

The second ontology is also a "valid" error: the ontology uses a synonym type `synonym` it has never defined. This is probably an issue caused by the OWL to OBO mapping: have you tried using the `envo.owl` instead ?

The third one is likely caused by the OBO parser, which I want to refactor anyway to be more compliant with the *1.4* syntax and semantics.

**Comment by jac2130** [comment]
Actually, I was able to make it all work.  If you want to I'd be happy to send you a pull request.  Since I'll be making some use of your code and I can benefit from refactoring, I might be able to help.  The first thing I'd do is get rid of the try and except blocks and then I'd make the loops recursive.  I'm a bit concerned with the memory usage.  While I was able to merge the envo with the sdgio ontologies, I was not able to merge in the global gazetteer because it just eats too much memory.  I can fix this by either getting more memory or by lowering the memory requirements of pronto or a combination of both perhaps.
